cmd/nitro/nitro.go

```diff
@@ -934,3 +941,80 @@ func testUpdateTxIndex(chainDb ethdb.Database, chainConfig *params.ChainConfig)
 	}
 	log.Info("Tx lookup entries written")
 }
+
+type LiveNodeConfig struct {
+	mutex  sync.RWMutex
+	args   []string
+	config *NodeConfig
+}
+
+func (c *LiveNodeConfig) Get() *NodeConfig {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	return c.config
+}
+
+func (c *LiveNodeConfig) set(config *NodeConfig) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	newConfig := *c.config
+
+	newConfig.Node.Sequencer.MaxBlockSpeed = config.Node.Sequencer.MaxBlockSpeed
```

Comment:
I'm personally in favor of just setting `c.config = config` and leaving it up to the user to only hot reload appropriate changes, but I'd like the rest of the team's thoughts on this.

Reply:
I suggest in a follow-up PR we use reflection to recursively annotate which fields are hot-swappable. This gets us both safety _and_ simplicity but requires familiarity with go reflection. Since I've done quite a bit of this, I'll happily PR that change and ask for both of y'all's reviews :)